assets/js/modules/analytics-4/components/common/PropertySelect.test.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * GA4 Property Select component tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import PropertySelect from './PropertySelect';
+import { MODULES_ANALYTICS, ACCOUNT_CREATE } from '../../../analytics/datastore/constants';
+import { STORE_NAME } from '../../datastore/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import * as fixtures from '../../datastore/__fixtures__';
+import * as analyticsFixtures from '../../../analytics/datastore/__fixtures__';
+import { fireEvent, act, render, muteFetch } from '../../../../../../tests/js/test-utils';
+
+const {
+	createProperty,
+	createWebDataStream,
+	properties,
+	webDataStreams,
+} = fixtures;
+const { accounts } = analyticsFixtures.accountsPropertiesProfiles;
+const accountID = createProperty._accountID;
+const propertyID = createWebDataStream._propertyID;
+
+const setupRegistry = ( { dispatch } ) => {
+	dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: 'http://example.com' } );
+	dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( STORE_NAME ).receiveGetProperties( properties, { accountID } );
+	dispatch( STORE_NAME ).finishResolution( 'getProperties', [ accountID ] );
+
+	dispatch( STORE_NAME ).receiveGetWebDataStreams( webDataStreams, { propertyID } );
+	dispatch( STORE_NAME ).finishResolution( 'receiveGetWebDataStreams', { propertyID } );
+};
+
+const setupEmptyRegistry = ( { dispatch } ) => {
+	dispatch( MODULES_ANALYTICS ).setSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( STORE_NAME ).receiveGetProperties( [], { accountID } );
+	dispatch( STORE_NAME ).finishResolution( 'getProperties', [ accountID ] );
+};
+
+const setupRegistryWithNoAccountID = ( registry ) => {
+	setupRegistry( registry );
+	registry.dispatch( STORE_NAME ).finishResolution( 'getProperties', [ ACCOUNT_CREATE ] );
+};
```

Comment:
This is a bit confusing since `setupRegistry` will set an `accountID` 🤔

Reply:
Looking at the original test, I think the intent was: 
- check it renders enabled when there is a valid account ID
- dispatch an action to set account ID to `ACCOUNT_CREATE`, which is a valid selection but an invalid ID
- check it now renders disabled.

https://github.com/google/site-kit-wp/blob/11b7cff0245e1eae5e3d2ffb085cdb0fceabeef5/assets/js/modules/analytics/components/common/PropertySelect.test.js#L106-L116

Perhaps there is a simpler way to go about it, I agree it doesn't read well. Happy to do as you advise.

On a related note: looking into this again, I now realise that the slightly unconventional use of `useSelect` [in the original component ](https://github.com/google/site-kit-wp/blob/11b7cff0245e1eae5e3d2ffb085cdb0fceabeef5/assets/js/modules/analytics/components/common/PropertySelect.js#L42)must've have been to do with this requirement because we only want to get properties etc if there is actually a valid accountID.
I've gone back and refactored this, but tried to keep it more in line with how we currently use `useSelect`. I've also added a story for this scenario as it wasn't immediately obvious to me.

It might be an idea to have a call and discuss if I'm still wide of the mark with all of this. In hindsight this was probably the wrong issue to pick as my first adventure with GA4 😂

Reply:
Just a note here about our slack conversation. I'm going to try to add this guard against no accountID in the resolver like in Analytics - maybe we can clear this up a bit.

Reply:
I've added a guard in the `getProperties` resolver as per our conversation on Slack, so was able to tidy up the use of `useSelect`, thank you for that.